### PR TITLE
feat(modal & button): adds the danger flavor for modals and buttons

### PR DESCRIPTION
**Pull request by davidmenendez**
Closes https://github.com/carbon-design-system/carbon-components-react/issues/664

Adds the danger primary button, tertiary button, and danger modal.

#### Changelog

**New**

* `Modal`
  * Add `danger` prop to display danger variant 
* `Modal-test` now includes danger variant-related tests
* `Button`
  * Extend `kind` prop to include `dangerPrimary` and `tertiary` variant
* `Button-test` now includes test for new `dangerPrimary` and `tertiary` variant

### CODE REVIEW

**Comment by joshblack** [comment]
Hi there! 👋 Thanks for taking the time to contribute. It'd be helpful before we review to fill out the changelog to call out things that are additions to the library, things that are updates, or the removal of things in the library. This helps us look for patterns that could influence which semver range the merge commit might need to trigger, and is especially helpful when it comes to refactoring/extending existing components. Thanks!

**Comment by davidmenendez** [comment]
updated

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -56,7 +58,14 @@ Button.propTypes = {
   className: PropTypes.string,
   disabled: PropTypes.bool,
   small: PropTypes.bool,
-  kind: PropTypes.oneOf(['primary', 'secondary', 'danger', 'ghost']).isRequired,
+  kind: PropTypes.oneOf([
+    'primary',
+    'secondary',
+    'danger',
+    'ghost',
+    'dangerPrimary',
```
Comment (joshblack) [inline src/components/Button/Button.js]: Should we do `primary--danger`? cc @carbon-design-system/ui am unsure what naming convention we want to do here.

Comment (tw15egan) [reply src/components/Button/Button.js]: `danger--primary` (matches the class) or `primary--danger` works for me

**Comment by davidmenendez** [comment]
alright

File: src/components/Button/Button.js (hunk 1 above)
Comment (joshblack) [reply src/components/Button/Button.js]: How about `danger--primary` since it matches the class right now?

Comment (davidmenendez) [reply src/components/Button/Button.js]: updated

**Pull request merged**